src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php

```diff
@@ -38,47 +38,82 @@
  */
 class TextDescriptor extends Descriptor
 {
+    private const VERB_COLORS = [
+        'ANY' => 'default',
+        'GET' => 'blue',
+        'HEAD' => 'magenta',
+        'OPTIONS' => 'blue',
+        'POST' => 'green',
+        'PUT' => 'yellow',
+        'PATCH' => 'yellow',
+        'DELETE' => 'red',
+    ];
+
     public function __construct(
         private ?FileLinkFormatter $fileLinkFormatter = null,
     ) {
     }
 
     protected function describeRouteCollection(RouteCollection $routes, array $options = []): void
     {
-        $showControllers = isset($options['show_controllers']) && $options['show_controllers'];
-
-        $tableHeaders = ['Name', 'Method', 'Scheme', 'Host', 'Path'];
-        if ($showControllers) {
-            $tableHeaders[] = 'Controller';
-        }
-
-        if ($showAliases = $options['show_aliases'] ?? false) {
-            $tableHeaders[] = 'Aliases';
-        }
+        $showAliases = $options['show_aliases'] ?? false;
+        $showControllers = $options['show_controllers'] ?? false;
 
         $tableRows = [];
+        $shouldShowScheme = false;
+        $shouldShowHost = false;
         foreach ($routes->all() as $name => $route) {
             $controller = $route->getDefault('_controller');
 
+            $scheme = $route->getSchemes() ? implode('|', $route->getSchemes()) : 'ANY';
+            $shouldShowScheme = $shouldShowScheme || 'ANY' !== $scheme;
+
+            $host = '' !== $route->getHost() ? $route->getHost() : 'ANY';
+            $shouldShowHost = $shouldShowHost || 'ANY' !== $host;
+
             $row = [
-                $name,
-                $route->getMethods() ? implode('|', $route->getMethods()) : 'ANY',
-                $route->getSchemes() ? implode('|', $route->getSchemes()) : 'ANY',
-                '' !== $route->getHost() ? $route->getHost() : 'ANY',
-                $this->formatControllerLink($controller, $route->getPath(), $options['container'] ?? null),
+                'Name' => $name,
+                'Methods' => $this->formatMethods($route->getMethods()),
```

Comment:
There is no way to disable colors which is BC break if someone depends on the output, no?

Reply:
Nope, currently there is no BC layer planned. As discussed in the thread we don't expect anyone using this output for automation (like scripting) and only for visual output.